x-pack/plugins/ingest_manager/server/services/agents/crud.ts

```diff
@@ -52,14 +55,22 @@ export async function listAgents(
     filters.push(ACTIVE_AGENT_CONDITION);
   }
 
-  const { saved_objects: agentSOs, total } = await soClient.find<AgentSOAttributes>({
+  let { saved_objects: agentSOs, total } = await soClient.find<AgentSOAttributes>({
     type: AGENT_SAVED_OBJECT_TYPE,
     filter: _joinFilters(filters),
     sortField,
     sortOrder,
     page,
     perPage,
   });
+  // filtering for a range on the version string will not work,
+  // nor does filtering on a flattened field (local_metadata), so filter here
+  if (showUpgradeable) {
```

Comment:
this is not ideal, but i decided to filter on the results of the soClient.find().  As we discussed earlier I was unable to filter through the flattened field `local_metadata` and even if I could I still need to use semver to compare `local_metadata.elastic.agent.version` which is a string so I could not do a range comparison in KQL. @jen-huang @nchaulet